coreneuron/nrniv/nrn_setup.cpp

```diff
@@ -1137,9 +1140,20 @@ void read_phase2(FileHandler& F, int imult, NrnThread& nt) {
         nmech = F.read_int();
         tml_index = new int[nmech];
         ml_nodecount = new int[nmech];
+        int diff_modfiles_count = 0;
         for (int i = 0; i < nmech; ++i) {
             tml_index[i] = F.read_int();
             ml_nodecount[i] = F.read_int();
+            auto found_different_mod_file =
+                std::find(different_mod_files.begin(), different_mod_files.end(), tml_index[i]);
+            if (found_different_mod_file != different_mod_files.end()) {
+                printf("Error: %s is different version of MOD file than the one used by NEURON!\n",
+                       nrn_get_mechname(tml_index[i]));
+                diff_modfiles_count++;
+            }
+        }
+        if (diff_modfiles_count > 0) {
```

Comment:
here you can print message from rank 0 : `Error : NEURON and CoreNEURON must use same mod files for compatibility, x different mod files found. Re-compile special and special-core!`

Reply:
Done :+1: